### PR TITLE
Add xdem

### PR DESCRIPTION
**Pull request by erikmannerfelt**
Checklist
- [X] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [X] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example).
- [X] Source is from official source.
- [X] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [X] If static libraries are linked in, the license of the static library is packaged.
- [X] Build number is 0.
- [X] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [X] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [X] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by erikmannerfelt** [comment]
@conda-forge-admin, please restart ci

Win+OSX work but Linux fails to import libGL when importing opencv. This must be a server-side problem, right?

```
Traceback (most recent call last):
  File "/home/conda/staged-recipes/build_artifacts/xdem_1622125678874/test_tmp/run_test.py", line 2, in <module>
    import xdem
  File "/home/conda/staged-recipes/build_artifacts/xdem_1622125678874/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_p/lib/python3.9/site-packages/xdem/__init__.py", line 1, in <module>
    from . import coreg, dem, examples, spatial_tools, spstats, volume
  File "/home/conda/staged-recipes/build_artifacts/xdem_1622125678874/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_p/lib/python3.9/site-packages/xdem/volume.py", line 7, in <module>
    import cv2
ImportError: libGL.so.1: cannot open shared object file: No such file or directory
```

**Comment by erikmannerfelt** [comment]
@conda-forge-admin, please ping team

Sorry guys, but I can't figure this out! How does one specify `opencv` as a dependency? It seems a server-side sub-dependency is missing.

**Comment by erikmannerfelt** [comment]
There we go! I had to add a `yum-requirements.txt` with the content:
```
mesa-libGL
```
Maybe this could be added to the conda-forge documentation, just like some `matplotlib`-specific instructions exist? I saw in previous PRs that this is a rather common problem.

File: recipes/xdem/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+  license_file: LICENSE
+  summary: Set of tools to manipulate Digital Elevation Models (DEMs)
+  doc_url: https://xdem.readthedocs.io
+  dev_url: https://github.com/GlacioHack/xdem
+
+extra:
+  recipe-maintainers:
+  - erikmannerfelt
+  - adehecq
+  - rhugonnet
```
Comment (ocefpaf) [inline recipes/xdem/meta.yaml:54]: @adehecq and @rhugonnet are you OK becoming maintainers here?

Comment (rhugonnet) [reply recipes/xdem/meta.yaml:54]: :+1:

Comment (adehecq) [reply recipes/xdem/meta.yaml:54]: Yes

**Pull request merged**